### DISCUSSION TITLE
A vault for Importing Azgaar's FMG Content into Obsidian (repo link included)

### DISCUSSION THREAD

**Discussion by buttonpushertv** [farling42/obsidian-import-json#70] [Show and tell]
@farling42  - I thought you'd like to see the results of your efforts put to work. My vault project relies heavily on your plugin. It makes the whole thing possible, actually!

I have created a template vault for importing content from an Azgaar's Fantasy Map Generator map into Obsidian notes for use in a campaign. It does this by importing from a Full JSON export of the FMG world. There are detailed instructions in the vault (and a tutorial video coming at some point).

Beware! This is not a trivial process. It's quite fiddly and things need to be done very precisely to get it to work. I've decided to open my repo up to the community to see what people think and solicit feedback from others who may want to contribute. I am very much an Intermediate Level coder. I clearly know enough to be dangerous, but I think I've figured out how to do things in some novel ways to move this FMG content to Markdown notes that allow further expansion and story-telling.

This repo is very much a work-in-progress, but I think it's gotten to a point where it could be put to use for worldbuilding and gameplay. There are still things to work on and pull in from the FMG JSON file. And, eventually, more RPG & campaign tools will be incorporated to round out the vault's usability overall. While it *could* be incorporated into another existing vault, it would *not* be a trivial exercise.

Check it out here & let me know what you think:
https://github.com/buttonpushertv/buttonpusherTTRPG

**Comment by farling42** [comment]
Have you investigated using the batch mode of the JSON/CSV Importer which was added in 0.37.0? It should change the 6 steps into 1 step.

If there are any fields which you would like added to the definition of each step, then please let me know.

**Reply by buttonpushertv** [reply]
I haven't had a chance to check it out. I'll try this week.

Looking at the instructions in the ReadMe, I don't see how to define template files for each of the batch passes. Is there a way to do that? Or do you place everything in a single template and then use @importBatchStep in some sort of routing to the appropriate section? If so, what is the command to end the processing in a given section? Or do you group the individual sections in functions and then just call them based on the value of @importBatchStep?

(Sorry, I'm rambling. On another project and just glanced at things with the importer & haven't messed with it yet.)

**Reply by farling42** [reply]
The batch processing requires a single template file.
There is an example of an if statement that can be used to choose the relevant part(s) of the template file.

**Comment by buttonpushertv** [comment]
Finally updated my scripts for importing an Azgaar's Fantasy map Generator Map from JSON into an Obsidian vault. This update now allows you to Batch Import the entire JSON file (in just a few steps). It's still not a "simple" process. There are instructions in the Vault on how to do it. Also it is compatible with ver. 1.99 of FMG, so it includes some of the recent re-works they've done.

You want to download "Wyrmling Version 0.6" - it should be the most recent release. Just download the zip file in the release, decompress it where Obisdian can see it, and open it to take a look.

https://github.com/buttonpushertv/buttonpusherTTRPG

**Comment by farling42** [comment]
It's good to see that the batch mode helps.
Good luck with further updates :-)
